framework/src/main/java/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java

```diff
@@ -2174,4 +2181,76 @@ private boolean isRelevantHelper(TypeMirror tm) {
                 throw new BugInCF("isRelevantHelper(%s): Unexpected TypeKind %s", tm, tm.getKind());
         }
     }
+
+    /**
+     * Return the type of the default value of the given type. The default value is 0, false, or
+     * null.
+     *
+     * @param typeMirror a type
+     * @return the annotated type of {@code type}'s default value
+     */
+    public AnnotatedTypeMirror getDefaultValueAnnotatedType(TypeMirror typeMirror) {
+        return getAnnotatedType(getDefaultValueTree(typeMirror));
+    }
+
+    /**
+     * Return a tree for the default value of the given type. The default value is 0, false, or
+     * null.
+     *
+     * @param typeMirror a type
+     * @return a tree for {@code type}'s default value
+     */
+    private static LiteralTree getDefaultValueTree(TypeMirror typeMirror) {
+        switch (typeMirror.getKind()) {
+            case BYTE:
+                return createJCLiteral(TypeTag.BYTE, (byte) 0, typeMirror);
+            case CHAR:
+                return createJCLiteral(TypeTag.CHAR, '\u0000', typeMirror);
+            case SHORT:
+                return createJCLiteral(TypeTag.SHORT, (short) 0, typeMirror);
+            case LONG:
+                return createJCLiteral(TypeTag.LONG, 0L, typeMirror);
+            case FLOAT:
+                return createJCLiteral(TypeTag.FLOAT, 0.0f, typeMirror);
+            case INT:
+                return createJCLiteral(TypeTag.INT, 0, typeMirror);
+            case DOUBLE:
+                return createJCLiteral(TypeTag.DOUBLE, 0.0d, typeMirror);
+            case BOOLEAN:
+                return createJCLiteral(TypeTag.BOOLEAN, false, typeMirror);
+            default:
+                return createJCLiteral(TypeTag.BOT, null, typeMirror);
+        }
+    }
+
+    /** The constructor for the JCLiteral class. */
+    private static Constructor<JCLiteral> jcLiteralConstructor;
+
+    {
+        try {
+            jcLiteralConstructor =
+                    JCLiteral.class.getDeclaredConstructor(TypeTag.class, Object.class);
+        } catch (NoSuchElementException | NoSuchMethodException e) {
+            throw new BugInCF(e);
+        }
+        jcLiteralConstructor.setAccessible(true);
+    }
+
+    /**
+     * Calls the protected JCLiteral constructor.
+     *
+     * @param typeTag the literal's type tag
+     * @param value the literal's value
+     * @param typeMirror the typeMirror for the literal
+     * @return a JCLiteral tree for the given type tag and value
+     */
+    private static JCLiteral createJCLiteral(TypeTag typeTag, Object value, TypeMirror typeMirror) {
```

Comment:
`org.checkerframework.javacutil.trees.TreeBuilder#buildLiteral` seems to build a literal tree from a given value.  

You should be able to do something like this:

```
    public static LiteralTree buildLiteral(Object value, ProcessingEnvironment env) {
        Context context = ((JavacProcessingEnvironment) env).getContext();
         TreeMaker maker = TreeMaker.instance(context);
        return maker.Literal(value);
    }
```

(This sort of method should be in TreeUtils.)

In case it's not easy to figure out.  Here are the imports for the code above:
```
import com.sun.tools.javac.tree.TreeMaker;
import javax.annotation.processing.ProcessingEnvironment;
import com.sun.tools.javac.util.Context;
import com.sun.tools.javac.processing.JavacProcessingEnvironment;
```